src/SpineAnimation.ts

```diff
@@ -133,61 +149,158 @@ export class SpineAnimation extends Script {
+    // @ts-ignore
+    const subRenderElementPool = engine._subRenderElementPool;
+    for (let i = 0, n = subMeshes.length; i < n; i++) {
+      let material = materials[i];
+      if (!material) {
+        continue;
+      }
+      if (material.destroyed || material.shader.destroyed) {
+        // @ts-ignore
+        material = this.engine._meshMagentaMaterial;
+      }
+
+      const subRenderElement = subRenderElementPool.get();
+      // @ts-ignore
+      subRenderElement.set(this, material, mesh._primitive, subMeshes[i]);
+      renderElement.addSubRenderElement(subRenderElement);
+    }
+    // @ts-ignore
+    context.camera._renderPipeline.pushRenderElement(context, renderElement);
+  }
+
+  /**
+   * @internal
+   */
+  // @ts-ignore
+  protected override _updateBounds(worldBounds: BoundingBox): void {
+    const bounds = worldBounds;
     const offset = this._tempOffset;
     const size = this._tempSize;
     const temp = this._tempArray;
     const zSpacing = this.setting?.zSpacing || 0.01;
     const skeleton = this._skeleton;
     skeleton.getBounds(offset, size, temp);
-    const drawOrder = skeleton.drawOrder;
-    bounds.min.set(offset.x, offset.y, 0);
-    bounds.max.set(
-      offset.x + size.x,
-      offset.y + size.y,
-      drawOrder.length * zSpacing
+    const { x, y } = offset;
+    bounds.min.set(x, y, 0);
+    bounds.max.set(x + size.x, y + size.y, skeleton.drawOrder.length * zSpacing);
+    BoundingBox.transform(
+      bounds,
+      this.entity.transform.worldMatrix,
+      bounds,
     );
   }
 
   /**
-   * Spine animation custom clone.
+   * @internal
    */
-  _cloneTo(target: SpineAnimation) {
-    target.setSkeletonData(this.skeletonData);
+  // @ts-ignore
+  override _cloneTo(target: SpineAnimation, srcRoot: Entity, targetRoot: Entity): void {
+    // @ts-ignore
+    super._cloneTo(target, srcRoot, targetRoot);
+    target.initialize(this._skeletonData);
     const _cloneSetting = { ...this.setting };
     target.setting = _cloneSetting;
   }
 
-  private _disposeCurrentSkeleton() {
+  /**
+   * @internal
+   */
+  protected override _onDestroy(): void {
+    const mesh = this._mesh;
+    if (mesh) {
+      mesh.destroyed || this._addResourceReferCount(mesh, -1);
+      this._mesh = null;
+    }
     this._skeletonData = null;
     this._skeleton = null;
-    this._state = null;
-  }
-
-  onDestroy() {
-    this._disposeCurrentSkeleton();
     this._meshGenerator = null;
     this.setting = null;
+    super._onDestroy();
+  }
+
+  private _setMesh(mesh: Mesh): void {
+    if (mesh.name === 'spine-mesh') {
+      this._mesh = mesh;
+    } else {
+      Logger.error('The SetMesh method of Spine Animation can only be called internally.');
+    }
   }
 }
+
+/**
+ * @remarks Extends `RendererUpdateFlag`.
+ */
+enum SpineAnimationUpdateFlags {
+  /** VertexElementMacro. */
+  VertexElementMacro = 0x2,
+}
+
+type RenderContext = { camera: Camera, _renderPipeline: BasicRenderPipeline };
```

Comment:
Can export from engine , or use any, and not _renderPipeline

Reply:
Discussed. Change to any.